speclang_modern.tex

```diff
@@ -3114,42 +3134,44 @@ \subsection{Model variables and model fields}
 Although the syntax of model variables is close to JML model
 variables, they differ in the sense that the type of a model variable
 is a logic type, not a C type. Also, the semantics above is closer to
-the one of B machines~\cite{abrial96:_b_book}. It has to be noticed
+the one of B machines~\cite{abrial96:_b_book}. It should be noticed
 that program verification with model variables does not have a
 well-established theoretical background~\cite{marche07,leavens07}, so
 we deliberately do not provide a precise semantics in this document .
 
+
+
 \section{Ghost variables and statements}
 \label{sec:ghost}
 \indextt{ghost}
 Ghost variables and statements are like C variables and statements,
 but visible only in the specifications. They are introduced by the
 \lstinline|ghost| keyword at the beginning of the annotation
 (i.e. \lstinline|/*@ ghost ... */| or \lstinline|//@ ghost ...|
-for a one-line
+for one-line
 ghost code, as mentioned in section~\ref{sec:gener-about-annot}).
 The grammar is given in Figure~\ref{fig:gram:ghost}, in which only the
 first form of annotation is used. In this figure, the \textit{C-*}
 non-terminals refer to the corresponding grammar rules of the ISO standard,
-without any ACSL extension. Any non terminal of the form
+without any ACSL extension. Any non-terminal of the form
 \textit{ghost-non-term} for which no definition is given in the figure
 represents the corresponding \textit{C-non-term} entry, in which any
 \textit{entry} is substituted by \textit{ghost-entry}.
 
 The variations with respect to the C
 grammar are the following:
 \begin{itemize}
-\item Comments must be introduced by \lstinline|//| and extend until the
+\item Comments within ghost code must be introduced by \lstinline|//| and extend until the
   end of the line (the ghost code itself is placed inside a C
-  comment. \lstinline|/* ... */| would thus lead to incorrect C code).
+  comment, so an embedded comment of the form \lstinline|/* ... */| would be incorrect C code).
 \item It is however possible to write multi-line annotations inside ghost
   code. These annotations are enclosed between
   \notimplemented{\lstinline|/@| and \lstinline|@/|}
   (since as indicated above, \lstinline|/*@ ... */| would lead
   to incorrect C code).
   As in normal annotations, \lstinline|@| characters at the beginning of a
   line and at the end of an annotation (before the final \lstinline|@/|) are
-  considered as blank.
+  considered as blank. \marginpar{Is this form of multi-line comments embedded in multi-line C (or ACSL) comments generally applicable?  If so, it should be introduced in section 1.2.}
```

Comment:
No, this is strictly restricted to ghost code, and this is not a plain comment but an ACSL annotation. Moreover, as we are not in _Inception_, you cannot have ghost code inside ghost code, which probably ought to be mentioned here.

Reply:
OK - added some explanation.